core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -181,11 +188,13 @@ class SocketServerTest extends JUnitSuite {
   def testSocketsCloseOnShutdown() {
     // open a connection
     val plainSocket = connect(protocol = SecurityProtocol.PLAINTEXT)
+    plainSocket.setTcpNoDelay(true)
     val traceSocket = connect(protocol = SecurityProtocol.TRACE)
+    traceSocket.setTcpNoDelay(true)
     val bytes = new Array[Byte](40)
     // send a request first to make sure the connection has been picked up by the socket server
-    sendRequest(plainSocket, bytes, Some(0))
-    sendRequest(traceSocket, bytes, Some(0))
+    sendRequest(plainSocket, bytes, Some(0), true)
+    sendRequest(traceSocket, bytes, Some(0), true)
```

Comment:
These requests are being sent while the socket is open. Did you mean to make the boolean true for the two requests that happen later?